### DISCUSSION TITLE
nanoid on prisma schema

### DISCUSSION THREAD

**Discussion by jlorezz** [prisma/prisma#12621] [Q&A]
Is there a way to generate nanoid() on default in the Prisma schema? `@default(nanoid())`

**Comment by nurul3101** [comment]
Hey @florianbress 👋 

`nanoid()` is not officially supported yet. But there is a workaround as mentioned [here](https://github.com/prisma/prisma/issues/2117#issuecomment-868728650) which should allow you to use nanoid.

Here's how you need to define your id.
```prisma
model test {
  id String @id @default(dbgenerated("nanoid()")) @db.VarChar(11)
}
```
Please note that you would need to add nanoid function to your database. 
Here's the link to [nanoid function](https://github.com/Jakeii/nanoid-postgres/blob/main/nanoid.sql) for PostgreSQL.

**Comment by jlorezz** [comment]
How is this done with Prisma, planet-scale, and MySQL?

**Reply by kierrereeg** [reply]
Did anyone ever figure this one out?

**Reply by okcha4ever** [reply]
I migrated my PostgreSQL database to supabase, since it has an sql editor

**Comment by Sirjoseph94** [comment]
Any chance of officially supporting this soon?

**Reply by janpio** [reply]
There is an open PR that needs are view from us. We'll try to get to it soon.

**Comment by inz-taylor** [comment]
This was added in [v4.9.0](https://github.com/prisma/prisma/releases/tag/4.9.0). I think this can be closed.

**Reply by janpio** [reply]
The issue was closed there, but nanoid is not supported yet.
Progress on this is now tracked in https://github.com/prisma/prisma/issues/18612

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
